sql/importer: TestRandomParquetExports failed

sql/importer.TestRandomParquetExports [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7599372?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7599372?buildTab=artifacts#/) on master @ [8357abb668a5adaff781343b394b162fb1b66c6e](https://github.com/cockroachdb/cockroach/commits/8357abb668a5adaff781343b394b162fb1b66c6e):


```
github.com/cockroachdb/cockroach/pkg/util/mon.(*BytesMonitor).Stop(...)
	github.com/cockroachdb/cockroach/pkg/util/mon/bytes_usage.go:420
github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Cleanup(0xc008fcce00, {0x6a10a90, 0xc001a4e000})
	github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:579 +0x27e
github.com/cockroachdb/cockroach/pkg/sql/rowflow.(*rowBasedFlow).Cleanup(0xc00ab2f608, {0x6a10a90?, 0xc001a4e000?})
	github.com/cockroachdb/cockroach/pkg/sql/rowflow/row_based_flow.go:442 +0x2e
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run.func2()
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:774 +0x2a
panic({0x4d868c0, 0xc008767b30})
	GOROOT/src/runtime/panic.go:890 +0x262
github.com/cockroachdb/cockroach/pkg/sql/importer.NewParquetColumn.func3({0x6a73300?, 0xc00ab2f728?})
	github.com/cockroachdb/cockroach/pkg/sql/importer/exportparquet.go:247 +0x70
github.com/cockroachdb/cockroach/pkg/sql/importer.NewParquetColumn.func47({0x6a72850?, 0xc008767b00})
	github.com/cockroachdb/cockroach/pkg/sql/importer/exportparquet.go:608 +0x188
github.com/cockroachdb/cockroach/pkg/sql/importer.(*parquetWriterProcessor).Run.func1(0xc008fcd200, {0x6a10b38, 0xc0087671a0}, 0xd?)
	github.com/cockroachdb/cockroach/pkg/sql/importer/exportparquet.go:789 +0xe2e
github.com/cockroachdb/cockroach/pkg/sql/importer.(*parquetWriterProcessor).Run(0xc008fcd200, {0x6a10a90?, 0xc001a4e000?})
	github.com/cockroachdb/cockroach/pkg/sql/importer/exportparquet.go:862 +0xd3
github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Run(0xc008fcce00, {0x6a10a90, 0xc001a4e000})
	github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:481 +0x254
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0xc0011e3cc0, {0x6a10b38, 0xc008766e40}, 0xc006898ee0, 0xc00af20bb0, 0xc0033f4680, 0xc004ee3c00, 0xc0030c2e80, 0x0)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:798 +0x955
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun(0xc008766ed0?, {0x6a10b38, 0xc008766e40}, 0x0?, 0xc006898ee0, 0x0?, {{0x6a129d8?, 0xc0064ba600?}, 0x0?}, 0xc004ee3c00)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1751 +0x1f9
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunAll(0xc006898ee0?, {0x6a10b38, 0xc008766e40}, 0xc0030c2b28?, 0x1?, 0xc0030c2b28, 0xc004ee3c00, 0xc0037dbe60?)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1503 +0x206
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine(0xc0030c2600, {0x6a10b38, 0xc008766e40}, 0xc0030c2b28, 0xc008766e40?, {0x7ff5f0f694c0?, 0xc00724c708}, 0x0?, 0xc005e59658)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1606 +0x4c8
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc0030c2600, {0x6a10b38, 0xc0077f11a0}, 0xc0030c2b28, {0x7ff5f0f694c0, 0xc00724c708})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1227 +0xbd9
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc0030c2600, {0x6a10b38?, 0xc0077f1080?}, {{0x6a39460, 0xc007138f40}, {0x0, 0x0, 0x0}, {0xc00517be03, 0x44}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:726 +0x2532
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt.func1({0x6a10b38?, 0xc0077f1080?})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:130 +0xbc
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithProfiling(0x6a10b38?, {0x6a10b38?, 0xc0077f1080?}, {0x6a39460?, 0xc007138f40?}, 0x0?, 0x96740a0?)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:2451 +0x330
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc0030c2600, {0x6a10b38, 0xc0077f1080}, {{0x6a39460, 0xc007138f40}, {0x0, 0x0, 0x0}, {0xc00517be03, 0x44}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:129 +0x4ba
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func1(0xc003e37ac0, 0xc0030c2600, 0xc003e377d8, 0xc003e37890, 0xc003e378d0, 0xc003e378b0, 0xc003e378a0, 0xc003e378c0)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1932 +0x305
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc0030c2600)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1937 +0x1127
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc0030c2600, {0x6a10a90, 0xc00487a1c0}, 0x41592d9340000000?, 0xc003d5bdd8?, 0xc00af68280?)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1857 +0x20b
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc0060ce640?, {0x6a10a90?, 0xc00487a1c0?}, {0xc001af3100?}, 0x3?, 0xc0069fe260?)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:831 +0xe6
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1()
	github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:728 +0x40d
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync
	github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:639 +0x22e
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomParquetExports.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-21638

Epic CRDB-11732

```
panic: interface conversion: tree.Datum is *tree.DOidWrapper, not *tree.DString
	panic: flow: unexpected 10240 leftover bytes [recovered]
	panic: flow: unexpected 10240 leftover bytes
```

--- a/pkg/sql/importer/exportparquet.go
+++ b/pkg/sql/importer/exportparquet.go
@@ -605,7 +605,7 @@ func NewParquetColumn(typ *types.T, name string, nullable bool) (ParquetColumn, 
 				if elt.ResolvedType().Family() == types.UnknownFamily {
 					// skip encoding the datum
 				} else {
-					el, err = grandChild.encodeFn(elt)
+					el, err = grandChild.encodeFn(tree.UnwrapDOidWrapper(elt))
 					if err != nil {
 						return col, err
 					}

--- a/pkg/sql/importer/exportparquet_test.go
+++ b/pkg/sql/importer/exportparquet_test.go
@@ -161,7 +161,7 @@ func validateDatum(t *testing.T, expected tree.Datum, actual tree.Datum, typ *ty
 		eArr := expected.(*tree.DArray)
 		aArr := actual.(*tree.DArray)
 		for i := 0; i < eArr.Len(); i++ {
-			validateDatum(t, eArr.Array[i], aArr.Array[i], typ.ArrayContents())
+			validateDatum(t, tree.UnwrapDOidWrapper(eArr.Array[i]), aArr.Array[i], typ.ArrayContents())
 		}
 	case types.DateFamily:
 		// pgDate.orig property doesn't matter and can cause the test to fail

